core/src/main/java/org/mule/runtime/core/api/scheduler/SchedulerConfig.java

```diff
@@ -53,23 +31,23 @@ public static SchedulerConfig config() {
   private final Integer maxConcurrentTasks;
   private final String schedulerPrefix;
   private final String schedulerName;
-  private final RejectionAction rejectionAction;
+  private final Boolean waitAllowed;
   private final Supplier<Long> shutdownTimeoutMillis;
 
   private SchedulerConfig() {
     this.maxConcurrentTasks = null;
     this.schedulerPrefix = null;
     this.schedulerName = null;
-    this.rejectionAction = DEFAULT;
+    this.waitAllowed = null;
     this.shutdownTimeoutMillis = () -> null;
   }
 
   private SchedulerConfig(Integer maxConcurrentTasks, String schedulerPrefix, String schedulerName,
-                          RejectionAction rejectionAction, Supplier<Long> shutdownTimeoutMillis) {
+                          Boolean waitAllowed, Supplier<Long> shutdownTimeoutMillis) {
```

Comment:
No way to use boolean or Optional here if really required instead of nullable Boolean?